olive/workflows/run/run.py

```diff
@@ -147,12 +147,24 @@ def run_engine(package_config: OlivePackageConfig, run_config: RunConfig, data_r
 
         AzureMLSystem.olive_config = run_config.to_json()
 
-    no_evaluation = (
+    # Register passes since we need to know whether they need to run on target
+    for pass_config in (run_config.passes or {}).values():
+        logger.debug("Registering pass %s", pass_config.type)
+        package_config.import_pass_module(pass_config.type)
+
+    target_not_used = (
```

Comment:
do we need only import passes if user specify pass_flows like https://github.com/microsoft/Olive/pull/1072/files#diff-e2990fcd6e151efbd0dd8ccd6817c2cdcdca63a2aea18404d8368df6a93a178aR129?

Reply:
In theory we only need to import passes that are being used by the workflows. 

So if no pass_flow give, it uses all passes. If pass_flows give, it uses the union of the flows.

Reply:
updated.